src/Api/Extension/ClientResourceCollectionExtension.php

```diff
@@ -52,6 +54,11 @@ private function addClientExternalLinksFilter(QueryBuilder $queryBuilder, string
             $queryBuilder->innerJoin(sprintf('%s.beneficiary', $rootAlias), $beneficiaryAlias);
         }
 
+        if (Note::class === $resourceClass) {
+            $beneficiaryAlias = 'beneficiaire';
+            $queryBuilder->innerJoin(sprintf('%s.beneficiaire', $rootAlias), $beneficiaryAlias);
+        }
+
```

Comment:
I think it could be
```

        if (is_subclass_of($resourceClass, DonneePersonnelle::class)) {
            $beneficiaryAlias = 'beneficiaire';
            $queryBuilder->innerJoin(sprintf('%s.beneficiaire', $rootAlias), $beneficiaryAlias);
        }
```
(maybe with Note/Document check for the moment)

Reply:
OR
```
        if (is_subclass_of($resourceClass, DonneePersonnelle::class)) {
            if (in_array($resourceClass, self::HANDLED_CLASSES)) {
                $beneficiaryAlias = 'beneficiaire';
                $queryBuilder->innerJoin(sprintf('%s.beneficiaire', $rootAlias), $beneficiaryAlias);
            }
        }
```

Reply:
Good idea ! :brain: